miner/block_builder.go

```diff
@@ -185,6 +195,29 @@ func (t *BlockBuilder) listenForTx() {
 	}
 }
 
+func (t *BlockBuilder) listenForAtx() {
+	t.Log.Info("start listening for atxs")
+	for {
+		select {
+		case <-t.stopChan:
+			return
+		case data := <-t.atxGossipChannel:
+			if data != nil {
+				x, err := mesh.BytesAsAtx(bytes.NewReader(data.Bytes()))
+				/*t.Log.With().Info("got new atx", log.String("sender", x., log.String("receiver", x.Recipient.String()),
+				log.String("amount", x.AmountAsBigInt().String()), log.Uint64("nonce", x.AccountNonce), log.Bool("valid", err != nil))*/
+				if err != nil {
+					t.Log.Error("cannot parse incoming TX")
+					data.ReportValidation(IncomingTxProtocol, false)
+					break
+				}
+				data.ReportValidation(IncomingTxProtocol, true)
```

Comment:
Where the ATX syntactic validity checks will happen?

Reply:
TBD, will add a comment